src/app/core/registry/dynamic-registry.ts

```diff
@@ -0,0 +1,66 @@
+import { Entity, EntityConstructor } from "../entity/model/entity";
+import { InjectionToken } from "@angular/core";
+import { ComponentType } from "@angular/cdk/overlay";
+import { OnInitDynamicComponent } from "../view/dynamic-components/on-init-dynamic-component.interface";
+
+export class Registry<T> {
```

Comment:
Did you consider extending the Map instead of using one? This way you wouldn't have to implement functions like `has()`.